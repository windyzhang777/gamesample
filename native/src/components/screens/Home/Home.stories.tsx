import React from 'react';
import {storiesOf} from '@storybook/react-native';
import Home from '.';
import Provider from '../../../utils/Provider';

storiesOf('Home Screen', module)
  .addDecorator((story) => <Provider story={story()} />)
  .add('Normal Navigation Item', () => {
    return <Home />;
  });
