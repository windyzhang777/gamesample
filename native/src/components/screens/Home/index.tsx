import React, {useState} from 'react';
import HomeShared, {HomeProps, HomeRenderProps} from 'shared/containers/screens/Home';
import NavigationSystem from '../../NavigationSystem';
import {View} from 'react-native';
import {useSelector} from 'react-redux';
import SetDisplayName from '../../SetDisplayName';

/**
 * This is a native version of rendering
 * @param {HomeProps} props
 */

export default function Home(props: HomeProps) {
  return (
    <HomeShared
      {...props}
      useSelector={useSelector}
      renderComponent={({displayName, logout, navigationMenu}: HomeProps & HomeRenderProps) => {
        if (!displayName) {
          return <SetDisplayName />;
        }
        return (
          <View style={{height: '100%', width: '100%'}}>
            <NavigationSystem
              topBarTitle={'TopBar Title'}
              useState={useState}
              navigationMenu={navigationMenu}
              logout={logout}
            />
          </View>
        );
      }}
    />
  );
}
