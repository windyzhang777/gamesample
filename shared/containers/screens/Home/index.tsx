import {RenderedComponent} from 'shared/models/RenderedComponent';
import {NavigationMenuType} from '../../../store/navigation-system/types';

/**
 * These are the properties of the HomeShared
 * These are passed to the dummy rendering component
 */
export interface HomeProps {
  useSelector?: any;
  useDispatch?: any;
  login?: () => void;
  logout?: () => void;
  displayName?: string;
  sessionTicket?: any;
}

export interface HomeRenderProps {
  navigationMenu: NavigationMenuType;
  login?: () => void;
  logout?: () => void;
}

/**
 * This is the navigation item's logic
 * @param {HomeProps} props properties for shared
 */
export default function Home({
  useSelector,
  login,
  renderComponent,
}: HomeProps & RenderedComponent<HomeProps & HomeRenderProps>) {
  /**
   * This is where we would implement our react logic
   * Such as using hooks
   */
  /**
   * This contains all the render properties
   * for what data we pass to be rendered
   */
  const renderProps: HomeProps & HomeRenderProps = {
    login,
    navigationMenu: useSelector((state: any) => state.navigationSystem),
    displayName: useSelector((state: any) => state.user.displayName),
    sessionTicket: useSelector((state: any) => state.user.sessionTicket),
  };
  return renderComponent(renderProps);
}
