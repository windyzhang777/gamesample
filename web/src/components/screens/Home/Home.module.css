.Neighborhood {
  width: 80%;
  margin: 60px auto 0;
  background-color: #faf4e0;
  height: calc(100vh - 60px);
  padding-top: 50vh;
  text-align: center;
  box-sizing: border-box;
  z-index: 90;
}
