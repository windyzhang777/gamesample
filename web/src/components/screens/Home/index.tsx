import React, {useState} from 'react';
import HomeShared, {HomeProps, HomeRenderProps} from 'shared/containers/screens/Home';
import NavigationSystem from '../../NavigationSystem';
import {useSelector} from 'react-redux';
import SetDisplayName from '../../SetDisplayName';

/**
 * This is a native version of rendering
 * @param {HomeProps} props
 */
export default function Home(props: HomeProps) {
  //This is just passing down the properties
  return (
    <HomeShared
      {...props}
      useSelector={useSelector}
      renderComponent={({
        login,
        displayName,
        navigationMenu,
        sessionTicket,
      }: HomeProps & HomeRenderProps) => {
        if (sessionTicket && !displayName) {
          return <SetDisplayName />;
        } 
        return (
          <NavigationSystem
            useState={useState}
            topBarTitle="westgate hills"
            navigationMenu={navigationMenu}
            login={login}
          />
        );
      }}
    />
  );
}
