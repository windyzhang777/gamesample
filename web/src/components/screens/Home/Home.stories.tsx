import React from 'react';
import {storiesOf} from '@storybook/react';
import Home from '.';
import Provider from '../../../utils/Provider';
import {useSelector} from 'react-redux';

storiesOf('Home Screen', module)
  .addDecorator((story) => <Provider story={story()} />)
  .add('Normal Navigation Item', () => {
    return <Home useSelector={useSelector} />;
  });
