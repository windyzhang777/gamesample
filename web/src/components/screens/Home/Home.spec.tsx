import React from 'react';
import {shallow} from 'enzyme';
import Home from './index';

describe('Home Component', () => {
  it('renders wrapper', () => {
    expect(shallow(<Home />)).toMatchSnapshot();
  });
  it('renders component without a display name', () => {
    const RenderPropComponent = shallow(<Home />).prop('renderComponent');
    expect(shallow(<RenderPropComponent />)).toMatchSnapshot();
  });
  it('renders component with a display name', () => {
    const RenderPropComponent = shallow(<Home />).prop('renderComponent');
    expect(
      shallow(
        <RenderPropComponent
          displayName="display name comes from redux via shared"
          logout="logout function generated from shared"
          navigationMenu="navigation menu from shared"
        />,
      ),
    ).toMatchSnapshot();
  });
});
