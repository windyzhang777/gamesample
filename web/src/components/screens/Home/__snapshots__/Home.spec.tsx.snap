// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component renders component with a display name 1`] = `
<NavigationSystem
  navigationMenu="navigation menu from shared"
  topBarTitle="westgate hills"
  useState={[Function]}
/>
`;

exports[`Home Component renders component without a display name 1`] = `
<NavigationSystem
  topBarTitle="westgate hills"
  useState={[Function]}
/>
`;

exports[`Home Component renders wrapper 1`] = `
<Home
  renderComponent={[Function]}
  useSelector={[Function]}
/>
`;
